### ISSUE TITLE
Unable to deploy Microsoft Teams Communicator App after changing App Registration passwords.

### ISSUE THREAD

**Issue by Schulro** [OfficeDev/microsoft-teams-apps-company-communicator#1074]
I am connecting the external GIT https://github.com/onemtc/microsoft-teams-apps-company-communicator
with the Branch mibon/imagetranslate and had to change the passwords of the App Registrations for CC User and CC Author because they where about the expire. 

I then created a new deployment and get the error only from the "Microsoft.Web/sites/sourcecontrols":
>There was a conflict. Conflict with existing ScmType: ExternalGit
{"code":"DeploymentFailed","target":"/subscriptions/xxxxx-fcf8-xxxx-9859-de3a40xx7f26ac/resourceGroups/fxxxx-TeamsCommunicator-ss/providers/Microsoft.Resources/deployments/Microsoft.Template-20230620165440","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.","details":[{"target":"/subscriptions/xxxxx-fcf8-xxxx-9859-de3a40xx7f26ac/resourceGroups/xxxx-TeamsCommunicator-sss/providers/Microsoft.Resources/deployments/Microsoft.Template-20230620165440","message":"There was a conflict. Conflict with existing ScmType: ExternalGit"},{"target":"/subscriptions/xxxxx-fcf8-xxxx-9859-de3a40xx7f26ac/resourceGroups/fxxxx-TeamsCommunicator-ss/providers/Microsoft.Resources/deployments/Microsoft.Template-20230620165440","message":"There was a conflict. Conflict with existing ScmType: ExternalGit"},{"target":"/subscriptions/xxxxx-fcf8-xxxx-9859-de3a40xx7f26ac/resourceGroups/fxxxx-TeamsCommunicator-ss/providers/Microsoft.Resources/deployments/Microsoft.Template-20230620165440","message":"There was a conflict. Conflict with existing ScmType: ExternalGit"},{"target":"/subscriptions/fxxxx-TeamsCommunicator-ss/resourceGroups/xxxx-TeamsCommunicator-sss/providers/Microsoft.Resources/deployments/Microsoft.Template-20230620165440","message":"There was a conflict. Conflict with existing ScmType: ExternalGit"}]}

After the secrets expire the application is not usable anymore and the deployment still fails after reconnecting the externalGIT.
Do you have an advice?
For more information please advice with TrackingID#2305120050003721

**Comment by gsv022** [comment]
Hi @Schulro ,
Thanks for raising the query.  If the secrets are about to expire or expired already.  Please refer to issue no:- [911](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/issues/911) for detailed steps on updating the new secret.

- As you have already went for new deployment, there is one resource or multiple resources got failed. Could you please share the screenshot of failed resources.
- **Note**:-  Furthermore, we recommend you to go with the standard version of Company Communicator. In this case we will be able to help you much better. We will not be able to provide much assistance on the issues which came on top of customized version. And do let us know if you have any issues with the standard version.

**Comment by Schulro** [comment]
![image](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/assets/137412615/1660379f-b350-40c2-bd9d-097357073733)
This is our current deployment.
We got the information from Microsoft to use this Branch as it has a translation function and you can add images.
If this is now in the default branch, we can also switch to this. Please guide me then in how to install it or can we just use the deployment button again and keep all current data and information?

As we do not have a key vault in our version, I can just edit the passwords in the configuration, what I already did.

**Comment by gsv022** [comment]
Hi @Schulro ,
If you wish to proceed with the standard/official version, kindly initiate a re-deploy and update the repository and branch while retaining all other parameters unchanged.

**Comment by Schulro** [comment]
Hi @gsv022,

we updated now to the main branch and fixed the deployment as well.
Now when going through the V4 Upgrade Guide we are not able to add the Application ID URI because it seems to be used from the AuthorApp. 
I checked in PowerShell and other Tools but cannot find anything there. Do you have an idea?

**Comment by gsv022** [comment]
Hi @Schulro ,
Since you have changed the repository and branch to the official one, I hope you have opted for a new deployment. Please correct me if I am mistaken.

**Comment by Schulro** [comment]
Hi @gsv022,

yes. New deployment has been done and was successfull. 
I am troubling at step 5. Set-up Authentication | Point 4.i. of the v5 Migration Guide.

Regards,
Robin

**Comment by gsv022** [comment]
Please provide supporting screenshots for further investigation

**Comment by Schulro** [comment]
So, after long time trying, I did the following.

- Remove the App Registration & Enterprise App for the Author in order to free the URL
- Created the App Registration & Enterprise App again
- Setup the Authentication according to Step 5 in the manual
- Updated all Deployments to External GIT and the Main branch and was finally able to perform a successfull deployment
- Updates the Author App according to the new Bot IDs and deleted the old one before uploading the new App

Now, whenever I try to Install the App in a MS Teams Team Channel (General) I get the error message "Something went wrong." and thats it.
Is there a log I can search for a more detailed error message to provide? Or do you have an idea?
I checked IDs now about 5 times, and I am sure they are correct.

Regards,
Robin

**Comment by gsv022** [comment]
- Please elucidate the reason for deleting the app registrations and creating new again. 
- Go to resource group-->Application Insights-->Under investigate click Failures. Copy the logs and paste it in this chat.
<img width="606" alt="image" src="https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/assets/98143492/bf54f9b0-7818-47a6-9df6-4f1c00e28713">

**Comment by Schulro** [comment]
I was not able to add the Application URI to the App Registration of the Graph App as it was already used by the Author App.
Therefore I had to delete the Author App. Right after the deletion I could do it.

![image](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/assets/137412615/4b7adfcc-0b40-4a62-a63d-ffbc3b1479c6)

Azure.RequestFailedException: The specified container already exists.
RequestId:124eb721-****-0077-40cf-aea217000000
Time:2023-07-04T23:30:01.2282552Z
Status: 409 (The specified container already exists.)
ErrorCode: ContainerAlreadyExists

Content:
<?xml version="1.0" encoding="utf-8"?><Error><Code>ContainerAlreadyExists</Code><Message>The specified container already exists.
RequestId:124eb721-****-0077-40cf-aea217000000
Time:2023-07-04T23:30:01.2282552Z</Message></Error>

Headers:
Server: Windows-Azure-Blob/1.0,Microsoft-HTTPAPI/2.0
x-ms-request-id: 124eb721-f01e-****-40cf-aea217000000
x-ms-client-request-id: 11fca9d4-a79f-****-ad89-bc36f2943521
x-ms-version: 2021-08-06
x-ms-error-code: ContainerAlreadyExists
Date: Tue, 04 Jul 2023 23:30:01 GMT
Content-Length: 230
Content-Type: application/xml

   at Azure.Storage.Blobs.ContainerRestClient.CreateAsync(Nullable`1 timeout, IDictionary`2 metadata, Nullable`1 access, String defaultEncryptionScope, Nullable`1 preventEncryptionScopeOverride, CancellationToken cancellationToken)
   at Azure.Storage.Blobs.BlobContainerClient.CreateInternal(PublicAccessType publicAccessType, IDictionary`2 metadata, BlobContainerEncryptionScopeOptions encryptionScopeOptions, Boolean async, CancellationToken cancellationToken, String operationName)

Command:
https://civ4sc***klrg.blob.core.windows.net/exportdatablobs?restype=container

**Comment by csnarain** [comment]
I had a pretty tough time installing this app using the MSFT provided ARM template. Finally after several rounds of trial and error, I finally managed to install it. I'm providing below a few things to be aware of in order to achieve a successful installation.

1. Ensure the 3 app registrations (graph, author, user) are brand new before you start the installation. Re-using ones from a previously failed attempt will break the deployment process
2. The GUIDs for service bus & storage account should be unique. The GUIDs used in the template are a good start but a failed installation will leave these GUIDs dangling and subsequent tries will fail. Anytime when you start a new installation ensure that you either remove the existing GUIDs (I haven't found a way to do it) or generate new GUIDs and use them. Please note that the GUIDs are unique to a particular AAD tenant
3. Another peculiar error I get is when we try to set the uri for app registration. If you have a previous unsuccessful attempt at the deployment, make sure you use a brand new base resource name when you re-attempt the deployment. Looks like AFD holds a registered domain even after deleting the resource (not sure for how long but it takes days for it to release it).
4. If you use a new GUID for the User App External ID item in the template, make sure you use the same GUID in the manifest file as well, else you'll get bizarre errors when adding the app in Teams.

Hope this helps.

**Comment by gsv022** [comment]
Hi @Schulro ,
Since authentication step has been done for author app instead of graph app, it is better to revert the changes for author and do the authentication for graph app. 

Please let us know the status

**Comment by Schulro** [comment]
Dear all,

since the above steps are basically a reinstallation instead of an update we deleted all ressources and set it up from the scretch.
Now everything is working again. Sadly we lost a functionality we had in another branch. There we had a button which translated the message in the CC, is there a posibility to place this as a feature request or add it manually?

Regards,
Robin

**Comment by gsv022** [comment]
@Schulro , The requested feature-_Button for translation_ has been added to backlog. Depending upon the priority, bandwidth and mainly post the discussion with internal team then it will be planned in the future releases of CC. Our support scope is very limited and adding the features additional to the official version will be considered as customized. Then we wont be able to assist much on that again.

Hope we conveyed. Thank you
